coalib/coala.py

```diff
@@ -59,11 +60,35 @@ def main(debug=False):
 
         if args.show_bears:
             from coalib.settings.ConfigurationGathering import (
-                get_filtered_bears)
-
-            local_bears, global_bears = get_filtered_bears(
-                args.filter_by_language, log_printer)
-
+                get_all_bears)
+            filtered_bears = get_all_bears(log_printer)
+            if args.filter_by_language:
+                # filter_by_language option should be deprecated soon.
+                # Since it is already covered by filter_by option.
+                log_printer.warn(
+                    "'--filter-by-language ...' option is deprecated. "
+                    "Use '--filter-by language ...' option instead")
+                args.filter_by = args.filter_by or []
```

Comment:
I can set default value while adding it in `DefaultArgParser.py`. But this is causing test failures in the following tests:
`tests/output/ConfWriterTest.py::ConfWriterTest::test_append`
`tests/output/ConfWriterTest.py::ConfWriterTest::test_write`

Because it assumes that config doesn't have `filter_by` option as it is not specified in this test.

Similarly, following two are also failing:
tests/settings/ConfigurationGatheringTest.py::ConfigurationGatheringTest::test_back_saving
tests/settings/ConfigurationGatheringTest.py::ConfigurationGatheringTest::test_gather_configuration